scripts/just-task.js

```diff
@@ -0,0 +1,32 @@
+// @ts-check
+
+const path = require('path');
+const { task, series, parallel, condition, option, argv } = require('just-task');
+
+require('./just-tasks/ts');
+require('./just-tasks/clean');
+require('./just-tasks/jest');
+require('./just-tasks/copy');
+require('./just-tasks/sass');
+require('./just-tasks/tslint');
+require('./just-tasks/webpack');
+
+option('production');
+option('min');
+
+task(
+  'build',
+  series(
+    'clean',
+    'copy',
+    parallel(
+      condition('tslint', () => !argv().min),
+      condition('jest', () => !argv().min),
+      series(
+        'sass',
+        parallel(condition('ts:commonjs', () => !argv().min), 'ts:esm', condition('ts:amd', () => argv().production && !argv().min)),
```

Comment:
it would be awesome if we can use commonJs even without min arg, since it is only needed for jest... (and mitigate using something like babel)

Even better if when used with this rig, jest remaps all module into the respective raw src folders (i.e. resolve into .ts files).

(But even as I type this, I can kind of foresee why this might actually cause jest to be slower since you've to recompile the TS files .. maybe that's more something for a different ("jest" build task))

Reply:
FWIW, WAC uses ts-jest as well. This is probably a necessary evil until esm works with jest natively. Then we can simply just do one compilation pass.